internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1459,6 +1471,21 @@ func resourceKubernetesClusterCreate(d *pluginsdk.ResourceData, meta interface{}
 		return err
 	}
 
+	autoUpgradeProfile := &managedclusters.ManagedClusterAutoUpgradeProfile{}
+	autoChannelUpgrade := d.Get("automatic_channel_upgrade").(string)
+	nodeOsChannelUpgrade := d.Get("node_os_channel_upgrade").(string)
+	if autoChannelUpgrade != "" {
+		if autoChannelUpgrade == string(managedclusters.UpgradeChannelNodeNegativeimage) && nodeOsChannelUpgrade != string(managedclusters.NodeOSUpgradeChannelNodeImage) {
```

Comment:
Looks like this wants to be 
```suggestion
		if autoChannelUpgrade == string(managedclusters.UpgradeChannelNodeNegativeimage) && string(managedclusters.NodeOSUpgradeChannelNodeImage) != "" && nodeOsChannelUpgrade != string(managedclusters.NodeOSUpgradeChannelNodeImage) {
```

to get around 
```
=== CONT  TestAccKubernetesCluster_upgradeChannel
testcase.go:110: Step 5/10 error: Error running apply: exit status 1
Error: `node_os_channel_upgrade` must be set to `NodeImage` if `automatic_channel_upgrade` is set to `node-image`
with azurerm_kubernetes_cluster.test,
on terraform_plugin_test.tf line 22, in resource "azurerm_kubernetes_cluster" "test":
22: resource "azurerm_kubernetes_cluster" "test" {
--- FAIL: TestAccKubernetesCluster_upgradeChannel (795.62s)
```